libcxx/modules/modules.json.in

```diff
@@ -0,0 +1,26 @@
+{
```

Comment:
How do you test that modules.json is a valid JSON file after substitution?

Reply:
Problems that can occur (not exhaustive):

- paths with `"` in them
- non-unicode
- control characters

Anything else should preserve JSON validity at least. I think these are all uncommon enough to ignore until someone complains and manages to convince a developer that `"<newline><invalid_unicode>` makes sens in an install directory.

Reply:
Then the contract is: along as the vendor behaves well, it is a valid json file.

Reply:
That's probably a better solution than yet another discussion about how paths aren't text and all the interesting ways they can't be encoded in valid UTF-8.

Reply:
I indeed don't want to test the path. In general we don't test our installation directly.

Reply:
@mordante That's only half true. We would want to tend towards testing our installation properties, however we don't have a great deal of infrastructure in place to do it. I did set up some tests in Lit here: [libcxx/test/libcxx/vendor/apple/system-install-properties.sh.cpp](https://github.com/llvm/llvm-project/blob/503f2ee4a86a740c4b4f442d95ff4e6d7f66a670/libcxx/test/libcxx/vendor/apple/system-install-properties.sh.cpp). It's just difficult to test in a portable way cause these things are usually inherently platform-specific.

Reply:
As mentioned during our private discussion. Thanks for the information.  I was not aware of this. Since we don't have a lot of infrastructure I don't feel we should block this patch on this testing.